### PR TITLE
En_Geg

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 1]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
+    this->actor.focus.rot.x = this->actor.world.rot.x;
+    this->actor.focus.rot.y = this->actor.world.rot.y;
+    this->actor.focus.rot.z = this->actor.world.rot.z;
+}
+
+s32 func_80BB1D64(EnGeg* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    Vec3f sp40;
+    Vec3f sp34;
+    s16 y = (this->actor.yawTowardsPlayer - this->unk_46E) - this->actor.shape.rot.y;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:351]: `y` along usually means y-position while this is dealing with angles and yaw. So you can rename this temp `yaw`

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 2]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
+        this->actionFunc = func_80BB27D4;
+        this->actor.flags &= ~0x10000;
+    } else if (this->actor.xzDistToPlayer < 150.0f) {
+        this->actor.flags |= 0x10000;
+        func_800B8614(&this->actor, globalCtx, 150.0f);
+    }
+}
+
+void func_80BB31B8(EnGeg* this, GlobalContext* globalCtx) {
+    s32 sp2C = 0x88;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:790]:
```suggestion
    s32 getItemId = GI_MASK_DON_GERO;
```
This is what this is

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 3]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
+
+    if (INV_CONTENT(ITEM_MASK_DON_GERO) == ITEM_MASK_DON_GERO) {
+        if (Player_GetMask(globalCtx) == PLAYER_MASK_DON_GEROS_MASK) {
+            this->unk_496 = 0xD8B;
+            sp2C = 5;
+        } else {
+            this->unk_496 = 0xD73;
+            sp2C = 5;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:798]:
```suggestion
            getItemId = GI_RUPEE_PURPLE;
        } else {
            this->unk_496 = 0xD73;
            getItemId = GI_RUPEE_PURPLE;
```

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 4]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.parent = NULL;
+        gSaveContext.weekEventReg[61] |= 1;
+        if (sp2C == 0x88) {
+            this->unk_230 |= 0x40;
+        }
+        this->actionFunc = func_80BB32AC;
+    } else {
+        func_800B8A1C(&this->actor, globalCtx, sp2C, 300.0f, 300.0f);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:812]:
```suggestion
        if (getItemId == GI_MASK_DON_GERO) {
            this->unk_230 |= 0x40;
        }
        this->actionFunc = func_80BB32AC;
    } else {
        func_800B8A1C(&this->actor, globalCtx, getItemId, 300.0f, 300.0f);
```

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 5]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
+}
+
+void EnGeg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnGeg* this = THIS;
+    s32 pad2;
+    s32 sp34[] = {
+        0x0000003E,
+        0x00000F64,
+    };
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:870]:
```suggestion
    s32 sp34[] = {
        0x3E,
        0xF64,
    };
```

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 6]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
+    MtxF sp28;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB1B14.s")
+    Matrix_MultiplyVector3fByState(&sp70, &sp7C);
+    Matrix_CopyCurrentState(&sp28);
+    func_8018219C(&sp28, &sp68, 0);
+    *arg2 = sp7C;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB1C1C.s")
+    if ((arg4 == 0) && (arg5 == 0)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:931]:
```suggestion
    if (!arg4 && !arg5) {
```
Both `arg4` and `arg5` look boolean from the code

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 7]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
+        arg3->x = sp68.x;
+        arg3->y = sp68.y;
+        arg3->z = sp68.z;
+        return true;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB1C8C.s")
+    if (arg5 != 0) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:938]:
```suggestion
    if (arg5) {
```

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 8]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
+    s32 phi_v1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB2A54.s")
+    switch (limbIndex) {
+        case 17:
+            if (this->unk_230 & 2) {
+                phi_v1 = 1;
+            } else {
+                phi_v1 = 0;
+            }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:998]:
```suggestion
            if (this->unk_230 & 2) {
                phi_v1 = true;
            } else {
                phi_v1 = false;
            }
```

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 9]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB2B1C.s")
+            if (this->unk_242 != 0) {
+                phi_v0 = 1;
+            } else {
+                phi_v0 = 0;
+            }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:1004]:
```suggestion
            if (this->unk_242 != 0) {
                phi_v0 = true;
            } else {
                phi_v0 = false;
            }
```

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 10]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
+            Matrix_StatePush();
+            break;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB2F7C.s")
+        case 10:
+            if (this->unk_230 & 2) {
+                phi_v1 = 1;
+            } else {
+                phi_v1 = 0;
+            }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:1023]:
```suggestion
            if (this->unk_230 & 2) {
                phi_v1 = true;
            } else {
                phi_v1 = false;
            }
```

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 11]
```diff
@@ -97,87 +146,950 @@ static DamageTable D_80BB3ED4 = {
...
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Geg/func_80BB30B4.s")
+            if (this->unk_242 != 0) {
+                phi_v0 = 1;
+            } else {
+                phi_v0 = 0;
+            }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:1029]:
```suggestion
            if (this->unk_242 != 0) {
                phi_v0 = true;
            } else {
                phi_v0 = false;
            }
```

**Review by engineer124 — COMMENTED** [review]
Good work! A few comments

**Review by AngheloAlf — APPROVED** [review]

**Review by engineer124 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 12]
```diff
@@ -97,87 +124,955 @@ static DamageTable D_80BB3ED4 = {
...
+}
+
+void EnGeg_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnGeg* this = THIS;
+    s32 pad2;
+    s32 sp34[] = {
+        0x3E,
+        0xF64,
+    };
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:852]: Could you remove the trailing comma and reformat?

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c [hunk 13]
```diff
@@ -97,87 +124,955 @@ static DamageTable D_80BB3ED4 = {
...
+        this->actor.child->scale.x *= 0.98f;
+        this->actor.child->scale.y *= 0.98f;
+        this->actor.child->scale.z *= 0.98f;
+
+        if (this->unk_4E0 > 70) {
+            for (i = 0; i < ARRAY_COUNT(this->unk_4C0); i++) {
+                this->unk_4C0[i].x *= temp_f20;
+                this->unk_4C0[i].y *= temp_f20;
+                this->unk_4C0[i].z *= temp_f20;
+                EffectSsHahen_Spawn(globalCtx, &this->unk_4B4, &this->unk_4C0[i], &sp74, 1, sp68, 1, 15, D_0406AB30);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Geg/z_en_geg.c:684]: `D_0406AB30` is from gameplay_keep. Should be able to include and use the symbol.

**Review by hensldm — COMMENTED** [review]
A couple things.

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c (hunk 13 above)
Comment (Kenix3) [reply src/overlays/actors/ovl_En_Geg/z_en_geg.c:684]: Looks like we already it.

File: src/overlays/actors/ovl_En_Geg/z_en_geg.c (hunk 12 above)
Comment (Kenix3) [reply src/overlays/actors/ovl_En_Geg/z_en_geg.c:852]: Agreed, but it doesn't need to hold up the merge.

**Review by Kenix3 — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Pull request merged**